### PR TITLE
Add UsageTracker gRPC service definition

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

Add UsageTracker gRPC service.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/usagetracker/usagetrackerpb/usagetracker.proto [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+service UsageTracker {
+  rpc TrackSeries(TrackSeriesRequest) returns (TrackSeriesResponse);
+}
+
+message TrackSeriesRequest {
+  // The tenant owning the series.
+  string userID = 1;
+
+  // The hashes of the series to track.
+  repeated int64 seriesHashes = 2;
```
Comment (colega) [inline pkg/usagetracker/usagetrackerpb/usagetracker.proto:23]: Can this be `uint64`?

Comment (colega) [inline pkg/usagetracker/usagetrackerpb/usagetracker.proto:23]: Looks good as a first approach, but given the horrible preformance of grpc unmarshaling of slices (it doesn't build the slice of a correct length), we may want to switch to just a []byte and read that from the service implementation in a streamed fashion.

That can be done in a future version though.

Comment (pracucci) [reply pkg/usagetracker/usagetrackerpb/usagetracker.proto:23]: I agree. Maybe we can keep this for a production version, rather in the prototype. Looks something we know how to optimize.

Comment (pracucci) [reply pkg/usagetracker/usagetrackerpb/usagetracker.proto]: Done in https://github.com/grafana/mimir/pull/10083/commits/98e96f4fb4a47b03beab5e7ff7111e9cb02e9104

**Review by colega — APPROVED** [review]

**Pull request merged**